plugin/pkg/auth/authorizer/node/node_authorizer_test.go

```diff
@@ -840,71 +847,41 @@ func populate(graph *Graph, nodes []*corev1.Node, pods []*corev1.Pod, pvs []*cor
 	}
 }
 
+func randomSubset(a, b int) []int {
+	return rand.Perm(b)[:a]
+}
+
 // generate creates sample pods and persistent volumes based on the provided options.
 // the secret/configmap/pvc/node references in the pod and pv objects are named to indicate the connections between the objects.
 // for example, secret0-pod0-node0 is a secret referenced by pod0 which is bound to node0.
 // when populated into the graph, the node authorizer should allow node0 to access that secret, but not node1.
-func generate(opts sampleDataOpts) ([]*corev1.Node, []*corev1.Pod, []*corev1.PersistentVolume, []*storagev1.VolumeAttachment) {
+func generate(opts *sampleDataOpts) ([]*corev1.Node, []*corev1.Pod, []*corev1.PersistentVolume, []*storagev1.VolumeAttachment) {
 	nodes := make([]*corev1.Node, 0, opts.nodes)
 	pods := make([]*corev1.Pod, 0, opts.nodes*opts.podsPerNode)
 	pvs := make([]*corev1.PersistentVolume, 0, (opts.nodes*opts.podsPerNode*opts.uniquePVCsPerPod)+(opts.sharedPVCsPerPod*opts.namespaces))
 	attachments := make([]*storagev1.VolumeAttachment, 0, opts.nodes*opts.attachmentsPerNode)
 
+	rand.Seed(12345)
+	if opts.sharedSecretsPerNamespace < opts.sharedSecretsPerPod {
+		opts.sharedSecretsPerNamespace = opts.sharedSecretsPerPod
```

Comment:
not a fan of mutating the passed-in options here... either use `max(shared*PerPod, shared*PerNamespace)` at point of use or revert the options to a struct so changes here are local

Reply:
done